galasa-managers-parent/galasa-managers-visualisation-parent/dev.galasa.elasticlog.manager/src/main/java/dev/galasa/elasticlog/internal/ElasticLogManagerImpl.java

```diff
@@ -0,0 +1,227 @@
+     * 
+     * @throws ManagerException
+     * @throws ResourceUnavailableException
+     */
+    @Override
+    public void provisionBuild() throws ManagerException, ResourceUnavailableException {
+        this.client = this.httpManager.newHttpClient();
+        this.gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create();
+        logger.info("ElasticLog Clients Initialised");
+    }
+
+    /**
+     * Test class result step, build and send the document request
+     * 
+     * @throws ManagerException
+     */
+    @Override
+	  	public void testClassResult(@NotNull String finalResult, Throwable finalException) throws ManagerException {
+	    //Record test information
+		this.runProperties.put("testCase", this.framework.getTestRun().getTestClassName());
+		this.runProperties.put("runId", this.framework.getTestRunName());
+	    this.runProperties.put("startTimestamp", Date.from(this.framework.getTestRun().getQueued()));
+	    this.runProperties.put("endTimestamp", Date.from(Instant.now()));
+	    this.runProperties.put("requestor", this.framework.getTestRun().getRequestor());
+	    this.runProperties.put("result", finalResult);
+	    
+	    String request = this.gson.toJson(this.runProperties);
+	    logger.info("Sending Run Request to ElasticLog Endpoint");
+	    logger.trace("Document Request -\n" + request);
+	    sendRunData(request);
+    }
+    
+    /**
+     * Create the required indexes and send document requests
+     * 
+     * @param request - document request json
+     * @throws ElasticLogManagerException
+     */
+    private void sendRunData(String request) throws ElasticLogManagerException {
+        //Register endpoint data as confidential
+    	String index = ElasticLogIndex.get();
+    	String endpoint = ElasticLogEndpoint.get();
+		ctf.registerText(index, "ElasticLog Index");
+		ctf.registerText(endpoint, "ElasticLog Endpoint");
+    	try {
+            //Create mapping Json
+    		String mapping = createIndexMapping();
+    		logger.trace("Index Mapping Request -\n" + mapping);
+            
+            //Set up http client for requests
+            client.setTrustingSSLContext();
+            client.addOkResponseCode(201);
+            client.setURI(new URI(endpoint));
+            
+            //Create index if not present
+    		HttpClientResponse<String> response = client.head(endpoint + "/" + index);
+    		if (response.getStatusCode() == 404) 
+                client.putJson(index, mapping, false);
+            
+            //Send document to index
+			client.postJson(index + "/_doc", request, false);
+        
+            //Create latest index if not present
+    		index = index + "_latest";
+    		HttpClientResponse<String> indexResponse = client.head(ElasticLogEndpoint.get() + "/" + index);
+			if (indexResponse.getStatusCode() == 404)
+                client.putJson(index, mapping, false);
+        
+            //Obtain id of document if test case is already recorded
```

Comment:
Why do we need to rewrite the test case?   What happens if 2 test runs are running at exactly the sametime against the same testcase.